common/browser.go

```diff
@@ -256,32 +256,11 @@ func (b *Browser) onAttachedToTarget(ev *target.EventAttachedToTarget) {
 	switch targetPage.Type {
 	case "background_page":
 		p, err := NewPage(b.ctx, session, browserCtx, targetPage.TargetID, nil, false, b.logger)
+		if err != nil && b.isPageAttachmentErrorIgnorable(ev, session, err) {
+			return // Ignore this page.
+		}
 		if err != nil {
-			isRunning := atomic.LoadInt64(&b.state) == BrowserStateOpen && b.IsConnected() // b.conn.isConnected()
-			if _, ok := err.(*websocket.CloseError); !ok && !isRunning {
-				// If we're no longer connected to browser, then ignore WebSocket errors
-				b.logger.Debugf("Browser:onAttachedToTarget:background_page:return", "sid:%v tid:%v websocket err:%v",
-					ev.SessionID, targetPage.TargetID, err)
-				return
-			}
-			select {
-			case <-b.ctx.Done():
-				b.logger.Debugf("Browser:onAttachedToTarget:background_page:return:<-ctx.Done",
-					"sid:%v tid:%v err:%v",
-					ev.SessionID, targetPage.TargetID, b.ctx.Err())
-				return // ignore
-			default:
-			}
-			// Another VU or instance closed the page, and the session is closed.
-			// This can happen if the page is closed before the attachedToTarget
-			// event is handled.
-			if session.Closed() {
-				b.logger.Debugf("Browser:onAttachedToTarget:return:session.Done",
-					"session closed: sid:%v tid:%v pageType:%s err:%v",
-					ev.SessionID, targetPage.TargetID, targetPage.Type, err)
-				return
-			}
-			k6ext.Panic(b.ctx, "creating a new background page: %w", err)
+			k6ext.Panic(b.ctx, "creating a new %s: %w", targetPage.Type, err)
```

Comment:
Could we return a `FatalError` here instead of panicking?

Reply:
Happy to help, but currently, making that change will require quite a few structural adjustments since it's called from a goroutine. However, let's discuss how to handle this kind of situation and work on it in another PR. Sound good? :)